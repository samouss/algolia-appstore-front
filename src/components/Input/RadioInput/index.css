.RadioInput {
  position: relative;
  overflow: hidden;
}

.RadioInput__Input {
  position: absolute;
  left: -99999px;

  &:checked + .RadioInput__Label .RadioInput__Radio__Inner {
    background-color: #0080ff;
  }
}

.RadioInput__Label {
  display: flex;
  align-items: center;
  width: 100%;
  padding: 5px 10px;

  &:hover {
    cursor: pointer;
    color: #0080ff;
  }
  &:hover > .RadioInput__Radio {
    border-color: #0080ff;
  }
}

.RadioInput__Radio {
  display: flex;
  width: 14px;
  height: 14px;
  border-radius: 100%;
  border: 1px solid;
  margin-right: 8px;
  margin-top: 1px;
  justify-content: center;
  align-items: center;
}

.RadioInput__Radio__Inner {
  width: 6px;
  height: 6px;
  border-radius: 100%;
}
