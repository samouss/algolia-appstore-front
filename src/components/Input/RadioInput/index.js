import React from 'react';
import PropTypes from 'prop-types';
import './index.css';

const RadioInput = ({
  id,
  name,
  value,
  checked,
  onChange,
  children,
  customClassName,
}) => (
  <div styleName="RadioInput">
    <input
      styleName="RadioInput__Input"
      id={id}
      name={name}
      value={value}
      type="radio"
      onChange={onChange}
      checked={checked}
    />
    <label
      htmlFor={id}
      className={customClassName}
      styleName="RadioInput__Label"
    >
      <span styleName="RadioInput__Radio">
        <span styleName="RadioInput__Radio__Inner" />
      </span>

      {children}
    </label>
  </div>
);

RadioInput.propTypes = {
  id: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  value: PropTypes.string.isRequired,
  checked: PropTypes.bool.isRequired,
  onChange: PropTypes.func.isRequired,
  children: PropTypes.node.isRequired,
  customClassName: PropTypes.string,
};

RadioInput.defaultProps = {
  customClassName: '',
};

export default RadioInput;
