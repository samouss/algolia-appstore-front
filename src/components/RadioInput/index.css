.RadioInput {
  position: relative;
  overflow: hidden;
}

.RadioInput__Input {
  position: absolute;
  left: -99999px;
}

.RadioInput__Label {
  width: 100%;
  padding: 5px 10px;

  &:hover {
    cursor: pointer;
    color: #0080ff;
  }
}

.RadioInput__Label--active {
  color: #0080ff;
}
