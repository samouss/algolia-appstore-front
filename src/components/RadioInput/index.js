import React from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';
import './index.css';

const RadioInput = ({
  name,
  value,
  checked,
  onChange,
  children,
  customClassName,
  customActiveClassName,
}) => (
  <div styleName="RadioInput">
    <input
      styleName="RadioInput__Input"
      id={`${name}-${value}`}
      name={name}
      type="radio"
      onChange={onChange}
      checked={checked}
    />
    <label
      htmlFor={`${name}-${value}`}
      className={cx(customClassName, {
        [customActiveClassName]: checked,
      })}
      styleName={cx('RadioInput__Label', {
        'RadioInput__Label--active': checked,
      })}
    >
      {children}
    </label>
  </div>
);

RadioInput.propTypes = {
  name: PropTypes.string.isRequired,
  value: PropTypes.string.isRequired,
  checked: PropTypes.bool.isRequired,
  onChange: PropTypes.func.isRequired,
  children: PropTypes.node.isRequired,
  customClassName: PropTypes.string,
  customActiveClassName: PropTypes.string,
};

RadioInput.defaultProps = {
  customClassName: '',
  customActiveClassName: '',
};

export default RadioInput;
