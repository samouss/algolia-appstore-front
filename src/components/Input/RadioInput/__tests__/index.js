import React from 'react';
import { shallow } from 'enzyme';
import RadioInput from '../index';

describe('<RadioInput />', () => {
  const defaultProps = {
    name: 'category',
    value: 'games',
    checked: false,
    onChange: () => {},
  };

  it('expect to render', () => {
    const props = {
      ...defaultProps,
    };

    const component = shallow(
      <RadioInput
        {...props}
      >
        label
      </RadioInput>,
    );

    expect(component).toMatchSnapshot();
  });

  it('expect to render when checked', () => {
    const props = {
      ...defaultProps,
      checked: true,
    };

    const component = shallow(
      <RadioInput
        {...props}
      >
        label
      </RadioInput>,
    );

    expect(component).toMatchSnapshot();
  });

  it('expect to render with customClassName', () => {
    const props = {
      ...defaultProps,
      customClassName: 'custom-class-name',
    };

    const component = shallow(
      <RadioInput
        {...props}
      >
        label
      </RadioInput>,
    );

    expect(component).toMatchSnapshot();
  });

  it('expect to render with customActiveClassName', () => {
    const props = {
      ...defaultProps,
      checked: true,
      customActiveClassName: 'custom-active-class-name',
    };

    const component = shallow(
      <RadioInput
        {...props}
      >
        label
      </RadioInput>,
    );

    expect(component).toMatchSnapshot();
  });

  it('expect to call onChange on input change', () => {
    const props = {
      ...defaultProps,
      onChange: jest.fn(),
    };

    const component = shallow(
      <RadioInput
        {...props}
      >
        label
      </RadioInput>,
    );

    component
      .find('input')
      .simulate('change');

    expect(props.onChange).toHaveBeenCalled();
  });
});
