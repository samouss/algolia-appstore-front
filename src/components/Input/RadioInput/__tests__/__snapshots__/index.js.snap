// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CheckboxInput /> expect to render 1`] = `
<div
  styleName="RadioInput"
>
  <input
    checked={false}
    id="category-games"
    name="category"
    onChange={[Function]}
    styleName="RadioInput__Input"
    type="radio"
    value="games"
  />
  <label
    className=""
    htmlFor="category-games"
    styleName="RadioInput__Label"
  >
    <span
      styleName="RadioInput__Radio"
    >
      <span
        styleName="RadioInput__Radio__Inner"
      />
    </span>
    label
  </label>
</div>
`;

exports[`<CheckboxInput /> expect to render when checked 1`] = `
<div
  styleName="RadioInput"
>
  <input
    checked={true}
    id="category-games"
    name="category"
    onChange={[Function]}
    styleName="RadioInput__Input"
    type="radio"
    value="games"
  />
  <label
    className=""
    htmlFor="category-games"
    styleName="RadioInput__Label"
  >
    <span
      styleName="RadioInput__Radio"
    >
      <span
        styleName="RadioInput__Radio__Inner"
      />
    </span>
    label
  </label>
</div>
`;

exports[`<CheckboxInput /> expect to render with customClassName 1`] = `
<div
  styleName="RadioInput"
>
  <input
    checked={false}
    id="category-games"
    name="category"
    onChange={[Function]}
    styleName="RadioInput__Input"
    type="radio"
    value="games"
  />
  <label
    className="custom-class-name"
    htmlFor="category-games"
    styleName="RadioInput__Label"
  >
    <span
      styleName="RadioInput__Radio"
    >
      <span
        styleName="RadioInput__Radio__Inner"
      />
    </span>
    label
  </label>
</div>
`;
